### ISSUE TITLE
unresolved-attribute: `PackageMetadata` has no attribute `get`

### ISSUE THREAD

**Issue by janosh** [astral-sh/ty#1134]
### Summary

```py
import importlib.metadata

package = 'importlib'
metadata = importlib.metadata.metadata(package)
homepage = metadata.get("version")
>>> error[unresolved-attribute] Type `PackageMetadata` has no attribute `get`
```

[not sure where that false positive is coming from](https://github.com/python/cpython/blob/e76464d161f2e7f2c27c84f8de4de5f88f025973/Lib/importlib/metadata/_meta.py#L24-L27)

### Version

ty 0.0.1-alpha.20 (f41f00af1 2025-09-03)

**Comment by sharkdp** [comment]
Thank you for reporting this.

I can not reproduce this: https://play.ty.dev/84bcd802-aa50-4100-88bd-721dcb3b9b0b

**Comment by sharkdp** [comment]
Probably need to configure your Python version to 3.12 or higher.

**Comment by carljm** [comment]
For the standard library, type checkers look at typeshed stubs, so it's more useful to look there than in the actual stdlib code to understand type checker behavior: https://github.com/python/typeshed/blob/main/stdlib/importlib/metadata/_meta.pyi#L22-L26

Will close this as "duplicate" of https://github.com/astral-sh/ty/issues/296, just to collect reports where that diagnostic improvement would have helped in finding the problem.

**Comment by janosh** [comment]
thanks @sharkdp, good catch! setting `requires-python = ">=3.12"` in `pyproject.toml` fixes the error

**Comment by sharkdp** [comment]
I meant to write a somewhat longer reply, but then I got a call and thought I'd send the short version anyway. Glad it helped.
